pkg/completion/completion.go

```diff
@@ -0,0 +1,113 @@
+package completion
+
+import (
+	"bufio"
+	"os"
+	"strings"
+
+	"github.com/spf13/cobra"
+)
+
+/* Autocomplete Functions for cobra ValidArgsFunction */
+
+// AutocompleteNamespaceFlag - Autocomplete the userns flag.
+// -> host, private, container, ns:[path], [path]
+func AutocompleteNamespaceFlag(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+	var completions []string
+	// If we don't filter on "toComplete", zsh and fish will not do file completion
+	// even if the prefix typed by the user does not match the returned completions
+	for _, comp := range []string{"host", "private", "container", "ns:"} {
+		if strings.HasPrefix(comp, toComplete) {
+			completions = append(completions, comp)
+		}
+	}
+	return completions, cobra.ShellCompDirectiveDefault
+}
+
+// autocompleteSubIDName - autocomplete the names in /etc/subuid or /etc/subgid
+// TODO: Move to c/common, podman also uses this
+func autocompleteSubIDName(filename string) ([]string, cobra.ShellCompDirective) {
+	file, err := os.Open(filename)
+	if err != nil {
+		return nil, cobra.ShellCompDirectiveError
+	}
+	defer file.Close()
+
+	var names []string
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		name := strings.SplitN(scanner.Text(), ":", 2)[0]
+		names = append(names, name)
+	}
+	if err = scanner.Err(); err != nil {
+		return nil, cobra.ShellCompDirectiveError
+	}
+
+	return names, cobra.ShellCompDirectiveNoFileComp
+}
+
+// AutocompleteSubgidName - Autocomplete subgidname based on the names in the /etc/subgid file.
+func AutocompleteSubgidName(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+	return autocompleteSubIDName("/etc/subgid")
+}
+
+// AutocompleteSubuidName - Autocomplete subuidname based on the names in the /etc/subuid file.
+func AutocompleteSubuidName(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+	return autocompleteSubIDName("/etc/subuid")
+}
+
+// AutocompleteCapabilities - Autocomplete linux capabilities options.
+// Used by --cap-add and --cap-drop; see: man capabilities
+// TODO: Move to c/common, podman also uses this
+func AutocompleteCapabilities(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+	LinuxCapabilities := []string{
+		"AUDIT_CONTROL",
```

Comment:
Might be able to share this list with containers/psgo/internal/capabilities/capabilities.go

Reply:
That would make sense, but psgo would need a way to export this list.